packages/controller/src/Controller.ts

```diff
@@ -361,9 +360,85 @@ export default class Controller {
 		}
 		await Promise.all(stopTasks);
 
+		logger.info("Saving data");
+		await this.saveData();
 		logger.info("Goodbye");
 	}
 
+	onAutosaveIntervalChanged() {
+		if (this.autosaveInterval) {
+			clearInterval(this.autosaveInterval);
+			this.autosaveInterval = undefined;
+		}
+		const autosaveIntervalSeconds = this.config.get("controller.autosave_interval");
+		if (autosaveIntervalSeconds > 0) {
```

Comment:
Is it documented that a negative or zero value results in no autosave?

Reply:
Why even allow those values? If the value is out of bounds - use default or produce an error.

You know the Merphy law, right? :)

Reply:
Lots of code needed to treat it specially. Need to get this project done...